### ISSUE TITLE
MVCC: Read uncommited

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sagarswathi/h2database#19]
```
What steps will reproduce the problem?

1. Save ReadOnlyCommitedERROR.java 
2. Run: javac -cp h2.jar ReadUncommitedERROR.java
3. Run: java -cp .;h2.jar ReadUncommitedERROR

What is the expected output? 

START
STOP

What do you see instead?

START
expected: Hello, got: Hallo
STOP

What version of the product are you using? On what operating system, file
system, and virtual machine?

H2 1.0.69
Windows XP SP2
java version "1.6.0_02"
Java(TM) SE Runtime Environment (build 1.6.0_02-b06)
Java HotSpot(TM) Client VM (build 1.6.0_02-b06, mixed mode, sharing)

Do you know a workaround?

No.

How important/urgent is the problem for you?

I think that it is critical.

In your view, is this a defect or a feature request?

This is defect. Definitely.

Please provide any additional information below.

Save ReadOnlyCommitedERROR.java 
Run: javac -cp h2.jar ReadOnlyCommitedOK.java
Run: java -cp .;h2.jar ReadOnlyCommitedOK

Result:
START
STOP

Differences in primary key key definition:
"PRIMARY KEY(ID, NAME)"
and
"PRIMARY KEY(ID)"

```

Original issue reported on code.google.com by `victor.p...@gmail.com` on 10 Apr 2008 at 7:24

**Comment by GoogleCodeExporter** [comment]
```
FIX :)

Save ReadOnlyCommitedOK.java
Run: javac -cp h2.jar ReadOnlyCommitedOK.java
Run: java -cp .;h2.jar ReadOnlyCommitedOK
```

Original comment by `victor.p...@gmail.com` on 10 Apr 2008 at 7:33

**Comment by GoogleCodeExporter** [comment]
```
Versions 1.0.68, 1.0.67 and 1.0.66 give same results.
```

Original comment by `victor.p...@gmail.com` on 11 Apr 2008 at 2:40

**Comment by GoogleCodeExporter** [comment]
```
When doing the UPDATE, the old row is first deleted, and the new row is then 
inserted.
The MVCC index thinks it's the same row (basically the same as DELETE + 
ROLLBACK)
because both the key is the same and the position in the data file is the same.
The solution is to use a 'row version' (a modification id), or not to re-use 
the same 
position in the data file when using MVCC.

I will try to fix this for the next release.
```

Original comment by `thomas.t...@gmail.com` on 12 Apr 2008 at 10:34
- Changed state: **Accepted**
- Added labels: **Priority-High**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
By the way, thanks a lot for the great test cases!
```

Original comment by `thomas.t...@gmail.com` on 12 Apr 2008 at 10:34

**Comment by GoogleCodeExporter** [comment]
```
Don't mention it. Thank you for your work! :)
```

Original comment by `victor.p...@gmail.com` on 14 Apr 2008 at 4:40

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 1.0.70
```

Original comment by `victor.p...@gmail.com` on 21 Apr 2008 at 7:09

**Comment by GoogleCodeExporter** [comment]
```
Fixed in version 1.0.70 (2008-04-20)
```

Original comment by `thomas.t...@gmail.com` on 21 Apr 2008 at 5:47
- Changed state: **Fixed**
